### DISCUSSION TITLE
i3 mode label when token is empty

### DISCUSSION THREAD

**Discussion by cstyan** [polybar/polybar#2795] [Q&A]
Sorry for the vague title, wasn't sure what would be better.

I have a module define that I'm using to display only the mode from my i3wm:

```
[module/i3-mode]
type = internal/i3
format = <label-mode>
format-foreground=${colors.alert}
label-mode=%mode:06:6%
```

I am setting all my modules to be in the center via `modules-center` and sizing their labels so that my bar modules don't move if the output of a script and it's label value change in length. However this does not seem to work for the built in i3 modules mode label. My assumption is that this is because the mode label can be empty if a binding mode is not currently being used. 

Is there any current way I can work around this behaviour? If not would a PR to optionally change the behaviour of the mode labels returned value be reasonable?

**Comment by patrick96** [comment]
I think a PR would be reasonable. Currently, `label-mode` is not used at all if the mode is `default`. The new option could just disable that check.

**Reply by cstyan** [reply]
This would be better than the current behaviour for my use case, but what I would prefer is that `label-mode` is returned with an empty string value. That way I can use the min/max width formatting options to ensure that the module still occupies the right amount of space, but nothing is displayed unless a mode other than `default` is active.

![2022-08-22-091920_561x73_scrot](https://user-images.githubusercontent.com/3246492/185974082-db51d491-3c9d-4d3b-b93d-54aa84fdcdb7.png)
![2022-08-22-092014_501x78_scrot](https://user-images.githubusercontent.com/3246492/185974094-e57a40f8-4035-41b3-8d3d-5024cf048366.png)

Because of the empty token, the current behaviour means that my modules are shifted around when a binding mode is active.

**Reply by patrick96** [reply]
I think that's reasonable as well. If people really want the `%mode%` token to show `default`, we can always add another token that behaves that way.
